micro-rdk-ffi/src/ffi/runtime.rs

```diff
@@ -226,24 +226,15 @@ pub unsafe extern "C" fn viam_server_start(ctx: *mut viam_server_context) -> via
         .unwrap();
     }
 
-    let max_connection = {
-        #[cfg(not(target_os = "espidf"))]
-        {
-            10
-        }
-        #[cfg(target_os = "espidf")]
-        {
-            use micro_rdk::esp32::esp_idf_svc::hal::sys::g_wifi_feature_caps;
-            use micro_rdk::esp32::esp_idf_svc::hal::sys::CONFIG_FEATURE_CACHE_TX_BUF_BIT;
-            if !g_spiram_ok {
-                log::info!("spiram not initialized disabling cache feature of the wifi driver");
-                g_wifi_feature_caps &= !(CONFIG_FEATURE_CACHE_TX_BUF_BIT as u64);
-                1
-            } else {
-                3
-            }
+    #[cfg(target_os = "espidf")]
```

Comment:
Logic to determine the number of max conns was present everywhere that led to instantiating a `ViamServerBuilder`, and was slightly different everywhere. I've opted to consolidate the logic inside `ViamServerBuilder` and let it decide the default. Explicit calls to `with_max_concurrent_connection` are no longer required to raise the default above one.